demos/projects/ESPRESSIF/esp32/config/demo_config.h

```diff
@@ -0,0 +1,174 @@
+ * 
+ * @warning If using X509 authentication, this MUST match the Common Name of the cert.
+ *
+ *  @note https://docs.microsoft.com/en-us/azure/iot-dps/concepts-service#registration-id
+ */
+#define democonfigREGISTRATION_ID CONFIG_AZURE_DPS_REGISTRATION_ID
+
+
+#endif // democonfigENABLE_DPS_SAMPLE
+
+/**
+ * @brief IoTHub device Id.
+ *
+ */
+#define democonfigDEVICE_ID CONFIG_AZURE_IOT_DEVICE_ID
+
+/**
+ * @brief IoTHub module Id.
+ * 
+ * @note This is optional argument for IoTHub
+ */
+#define democonfigMODULE_ID CONFIG_AZURE_IOT_MODULE_ID
+
+/**
+ * @brief IoTHub hostname.
+ *
+ */
+#define democonfigHOSTNAME CONFIG_AZURE_IOT_HUB_FQDN
+
+/**
+ * @brief Device symmetric key
+ *
+ */
+#define democonfigDEVICE_SYMMETRIC_KEY CONFIG_AZURE_IOT_DEVICE_SYMMETRIC_KEY
+
+/**
+ * @brief Client's X509 Certificate.
+ *
+ */
+// #define democonfigCLIENT_CERTIFICATE_PEM    "<YOUR DEVICE CERT HERE>"
+
+/**
+ * @brief Client's private key.
+ * 
+ */
+// #define democonfigCLIENT_PRIVATE_KEY_PEM    "<YOUR DEVICE PRIVATE KEY HERE>"
+
+/**
+ * @brief Baltimore Trusted Root CA.
+ *
+ */
+#define democonfigROOT_CA_PEM "-----BEGIN CERTIFICATE-----\r\n" \
+"MIIDdzCCAl+gAwIBAgIEAgAAuTANBgkqhkiG9w0BAQUFADBaMQswCQYDVQQGEwJJ\r\n" \
+"RTESMBAGA1UEChMJQmFsdGltb3JlMRMwEQYDVQQLEwpDeWJlclRydXN0MSIwIAYD\r\n" \
+"VQQDExlCYWx0aW1vcmUgQ3liZXJUcnVzdCBSb290MB4XDTAwMDUxMjE4NDYwMFoX\r\n" \
+"DTI1MDUxMjIzNTkwMFowWjELMAkGA1UEBhMCSUUxEjAQBgNVBAoTCUJhbHRpbW9y\r\n" \
+"ZTETMBEGA1UECxMKQ3liZXJUcnVzdDEiMCAGA1UEAxMZQmFsdGltb3JlIEN5YmVy\r\n" \
+"VHJ1c3QgUm9vdDCCASIwDQYJKoZIhvcNAQEBBQADggEPADCCAQoCggEBAKMEuyKr\r\n" \
+"mD1X6CZymrV51Cni4eiVgLGw41uOKymaZN+hXe2wCQVt2yguzmKiYv60iNoS6zjr\r\n" \
+"IZ3AQSsBUnuId9Mcj8e6uYi1agnnc+gRQKfRzMpijS3ljwumUNKoUMMo6vWrJYeK\r\n" \
+"mpYcqWe4PwzV9/lSEy/CG9VwcPCPwBLKBsua4dnKM3p31vjsufFoREJIE9LAwqSu\r\n" \
+"XmD+tqYF/LTdB1kC1FkYmGP1pWPgkAx9XbIGevOF6uvUA65ehD5f/xXtabz5OTZy\r\n" \
+"dc93Uk3zyZAsuT3lySNTPx8kmCFcB5kpvcY67Oduhjprl3RjM71oGDHweI12v/ye\r\n" \
+"jl0qhqdNkNwnGjkCAwEAAaNFMEMwHQYDVR0OBBYEFOWdWTCCR1jMrPoIVDaGezq1\r\n" \
+"BE3wMBIGA1UdEwEB/wQIMAYBAf8CAQMwDgYDVR0PAQH/BAQDAgEGMA0GCSqGSIb3\r\n" \
+"DQEBBQUAA4IBAQCFDF2O5G9RaEIFoN27TyclhAO992T9Ldcw46QQF+vaKSm2eT92\r\n" \
+"9hkTI7gQCvlYpNRhcL0EYWoSihfVCr3FvDB81ukMJY2GQE/szKN+OMY3EU/t3Wgx\r\n" \
+"jkzSswF07r51XgdIGn9w/xZchMB5hbgF/X++ZRGjD8ACtPhSNzkE1akxehi/oCr0\r\n" \
+"Epn3o0WC4zxe9Z2etciefC7IpJ5OCBRLbf1wbWsaY71k5h+3zvDyny67G7fyUIhz\r\n" \
+"ksLi4xaNmjICq44Y3ekQEe5+NauQrz4wlHrQMz2nZQ/1/I6eYs9HRCwBXbsdtTLS\r\n" \
+"R9I4LtD+gdwyah617jzV/OeBHRnDJELqYzmp\r\n" \
+"-----END CERTIFICATE-----\r\n"
+
+/**
+ * @brief Set the stack size of the main demo task.
+ *
+ */
+#define democonfigDEMO_STACKSIZE CONFIG_AZURE_TASK_STACKSIZE
+
+/**
+ * @brief Size of the network buffer for MQTT packets.
+ */
+#define democonfigNETWORK_BUFFER_SIZE CONFIG_NETWORK_BUFFER_SIZE
+
+/**
+ * @brief IoTHub endpoint port.
+ */
+#define democonfigIOTHUB_PORT CONFIG_AZURE_IOTHUB_PORT
+
+#define configRAND32() (rand()/RAND_MAX)
```

Comment:
>configRAND32 [](http://example.com/codeflow?start=8&length=12)

why do we need #include<stdlib.h>?

Reply:
FreeRTOSConfig.h doesn't seem to be used. If I define it there, it fails to compile. I'll check with you offline.

Reply:
Indeed config/FreeRTOSConfig.h is not being used. esp-idf was getting it from another location.

Reply:
Please see my comment below. I'll leave that last comment unresolved for reference.